File: energonsoftware/src/app/core/wow.ts
//#region Types

export interface IAddon {
  name: string;
  version: string;
  latest_version: boolean;
  enabled: boolean;
  url: string;
  description: string;
}

export interface IMacro {
  name: string;
  macro: string;
}

export interface IMacroClass {
  character_class: string;
  macros: IMacro[];
}

export interface IScreenshots {
  images: string[];
  text: string;
}

//#endregion

//#region Messages

export interface IGetAddonsResponse {
  wow_version: string;
  addons: IAddon[];
}

export interface IGetMacrosResponse {
  macro_classes: IMacroClass[];
}

export interface IGetScreenshotsResponse {
  images: IScreenshots[];
}

//#endregion

File: api/src/models/wow.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Default, Clone, Deserialize, Serialize)]
pub struct Addon {
    name: String,
    version: String,
    latest_version: bool,
    enabled: bool,
    url: String,
    description: String,
}

#[derive(Debug, Default, Clone, Deserialize, Serialize)]
pub struct Macro {
    name: String,
    r#macro: String,
}

#[derive(Debug, Default, Clone, Deserialize, Serialize)]
pub struct MacroClass {
    character_class: String,
    macros: Vec<Macro>,
}

#[derive(Debug, Default, Clone, Deserialize, Serialize)]
pub struct Screenshots {
    // TODO: should have per-image alt text
    images: Vec<String>,

    #[serde(default)]
    text: String,
}
